src/AmqpConnectionManager.ts

```diff
@@ -143,19 +153,20 @@ export default class AmqpConnectionManager extends EventEmitter implements IAmqp
         | (() => Promise<ConnectionUrl | ConnectionUrl[]>);
     private _urls?: ConnectionUrl[];
 
-    /**
-     * Keep track of whether a disconnect event has been sent or not.  The problem
-     * is that if we've never connected, and we encounter an error, we want to
-     * generate a "disconnect" event, even though we're not disconnected, otherwise
-     * the caller will never know there was an error.  So we can't just rely on
-     * this._currentConnection.
-     */
-    private _disconnectSent = false;
-
     public connectionOptions: AmpqConnectionOptions | undefined;
     public heartbeatIntervalInSeconds: number;
     public reconnectTimeInSeconds: number;
 
+    private _connectionAttempts = 0;
+
+    /**
+     * The number of connection attempts this connection manager has made,
+     * successful, failed, or in-progress..
+     */
+    get connectionAttempts(): number {
+        return this._connectionAttempts;
```

Comment:
Should we maybe rename it to `pendingConnectionAttempts` etc. and reset it on connect? Not sure ever increasing counter has sense?

Reply:
`pendingConnectionAttempts` makes it sound like it should either be 1 or 0 (we're trying to connect or we're not).

`connectionAttemptsSinceLastConnect`?  That's a bit wordy.

Reply:
I'll take the connection attempt count out for now - you can always track it yourself from the event if you really need it, and then we can get this merged.

Reply:
That is perfectly fine for me, actually that's how I was doing it till now, so just need to change the listened event. Thank you